lib/datadog/appsec.rb

```diff
@@ -21,3 +31,5 @@ def self.writer
 require_relative 'appsec/contrib/rack/integration'
 require_relative 'appsec/contrib/sinatra/integration'
 require_relative 'appsec/contrib/rails/integration'
+
+require_relative 'appsec/autoload'
```

Comment:
I guess the part that is not necessary anymore is the `require 'ddtrace'` within the autoload file. Since we could assume `ddtrace` has been loaded before 😄

Reply:
@GustavoCaso I presume you're correct.

That said while I don't think `require 'datadog/appsec/autoload'` was ever exposed to end users we should be extra careful and check the docs if that's the case anywhere.

Reply:
Also now that we don't gate out requiring `datadog/appsec`, I think `lib/datadog/appsec/autoload.rb` should ultimately go away, and `DD_APPSEC_ENABLED=true` should be handled via the regular configuration only.

There would be a bunch of questions remaining regarding auto-instrumenting AppSec as it is done currently when `DD_APPSEC_ENABLED=true` but not when `Datadog.configure { |c| c.appsec.enabled = true }`.